DependencyInjection/Configuration.php

```diff
@@ -185,136 +186,280 @@ private function configureClients(ArrayNodeDefinition $root)
                         ->defaultFalse()
                         ->info('Set to true to get the client wrapped in a BatchClient which allows you to send multiple request at the same time.')
                     ->end()
-                    ->arrayNode('plugins')
-                        ->info('A list of service ids of plugins. The order is important.')
-                        ->prototype('scalar')->end()
-                    ->end()
                     ->variableNode('config')->defaultValue([])->end()
-                ->end()
-            ->end();
+                    ->arrayNode('plugins')
+                        ->info('A list of plugin service ids and client specific plugin definitions. The order is important.')
+                        ->prototype('array')
+        ;
```

Comment:
You do not call `->end()` here and I know that it is on purpose. You should at least add a comment here. Or event better: Create the plugin node and then add it to the tree. (I think that is possible)